Update binary logger docs (#11369)
MicrosoftDocs/visualstudio-docs: docs/msbuild/obtaining-build-logs-with-msbuild.md

```cmd
-bl:binarylogfilename.binlog
```
You can use the [Structured log viewer](https://msbuildlog.com/) to view binary logs in a way that makes it much easier to see what is going on.

For more information, see [Command-line reference](../msbuild/msbuild-command-line-reference.md).

The binary log includes the following information from a build:
- Logged messages with the `diagnostic` verbosity level

> - The full contents of files imported by each project

Reviewer: @jeffkl Can you evaluate Copilot's suggestions here?

> - The full contents of files imported by each project

Author: I'd say its subjective. Build events means less to me personally than messages. Warnings and errors are just messages. I don't think the binary log contains "diagnostic MSBuild-engine information" either. You can set a different env var to get that sort of logging.

https://learn.microsoft.com/en-us/visualstudio/ide/msbuild-logs?view=visualstudio#capture-binary-logs-through-visual-studio

But just adding the binary logger from the command-line just gives you a binary log, setting the  `MSBuildDebugEngine` env var gives you a binary log and MSBuild engine diagnostic information. This part of the docs as far as I know is just talking about getting a binary log through a command-line argument.

- Properties, their final value, and any property reassignments
- Items, their metadata, and any item updates or removals
- Task inputs and outputs

> [!NOTE]

> > Since the binary logger includes a substantial amount of information from the build, large builds can experience measurable overhead when it is enabled.

Reviewer 2: ```suggestion
> Because the binary logger includes a substantial amount of information from the build, large builds can experience measurable overhead when it's enabled.
```
